doc/conf.py

```diff
@@ -28,14 +28,24 @@
 
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
-extensions = ['matplotlib.sphinxext.mathmpl', 'sphinxext.math_symbol_table',
-              'sphinx.ext.autodoc', 'matplotlib.sphinxext.only_directives',
-              'sphinx.ext.doctest', 'sphinx.ext.autosummary',
-              'sphinx.ext.inheritance_diagram', 'sphinx.ext.intersphinx',
-              'sphinx_gallery.gen_gallery',
-              'matplotlib.sphinxext.plot_directive',
-              'sphinxext.github',
-              'numpydoc']
+extensions = [
+    'sphinx.ext.autodoc',
+    'sphinx.ext.autosummary',
+    'sphinx.ext.doctest',
+    'sphinx.ext.inheritance_diagram',
+    'sphinx.ext.intersphinx',
+    'IPython.sphinxext.ipython_console_highlighting',
+    'IPython.sphinxext.ipython_directive',
+    'numpydoc',  # Needs to be loaded *after* autodoc.
+    'sphinx_gallery.gen_gallery',
+    'matplotlib.sphinxext.mathmpl',
+    'matplotlib.sphinxext.only_directives',
+    'matplotlib.sphinxext.plot_directive',
+    'sphinxext.github',
+    'sphinxext.math_symbol_table',
+    'sphinxext.mock_gui_toolkits',
```

Comment:
If we drop support for sphinx older than 1.3 we can probably replace this with the build in autodoc_mock_imports in  http://www.sphinx-doc.org/en/stable/ext/autodoc.html#confval-autodoc_mock_imports If we drop versions below 1.6 we can make it simpler by only mocking the top level modules. As I recall it I added the mocking here predates Sphinx 1.3 which is why its done manually

Reply:
What is the argument for not requiring 1.6?

Reply:
The only one I can think of is building with Sphinx from apt-get/yum but I am not sure we need to support that

Reply:
I think debian does build our docs as part of the build process. @sandrotosi  How much pain would it cause you to bump to sphinx 1.6?

Reply:
it would be no problem at all, actually it would be really helpful as in Debian we're migrating to sphinx 1.6 and the current doc cannot be built with that version, so thanks for pushing to use 1.6 :)

Reply:
@sandrotosi you might have to report an issue for that. Currently the docs are being build with the latest release (1.6.5) in CI so if something is not working for you with 1.6.x we might need to fix that independently

Reply:
nope, what i really should do is updating matplotlib in debian (which i'll do as soon as i find time for it)

Reply:
I don't think that's going to work because we need some mocked objects to be *types* (so that we can subclass them) or *callables* with specific return values (`sip.getapi`), rather than just "existing" (in other words, because we are running code at the toplevel rather than just defining a bunch of functions).

Reply:
Debian 7 “Wheezy” is still supported (https://wiki.debian.org/LTS) and it has `1.1.3` sphinx (https://packages.debian.org/search?keywords=python-sphinx) =\

Reply:
TBH i wouldnt worry about wheezy (nor jessie) - if anyone will want to get an updated mpl on that release, they'll have to take care of backporting a tons of other deps (numpy in primis) before caring about sphinx and missing documentation

Reply:
True, but even stretch does not have 1.6